File: isegoria/src/main/java/com/eulersbridge/isegoria/models/GenericUser.java
package com.eulersbridge.isegoria.models;

import android.support.annotation.Nullable;

import com.squareup.moshi.Json;

public class GenericUser {

    public String gender;
    public String nationality;

    public String email;
    public String givenName;
    public String familyName;

    @Nullable
    public Long institutionId;

    public long level;
    public long experience;

    @Json(name = "numOfCompTasks")
    public long completedTasksCount;

    @Json(name = "numOfCompBadges")
    public long completedBadgesCount;

    @Json(name = "profilePhoto")
    public String profilePhotoURL;

    public String getFullName() {
        return String.format("%s %s", givenName, familyName);
    }

    GenericUser() {
        // Required empty constructor
    }

    GenericUser(GenericUser user) {
        this.gender = user.gender;
        this.nationality = user.nationality;
        this.email = user.email;
        this.givenName = user.givenName;
        this.familyName = user.familyName;
        this.institutionId = user.institutionId;
        this.level = user.level;
        this.experience = user.experience;
        this.completedTasksCount = user.completedTasksCount;
        this.completedBadgesCount = user.completedBadgesCount;
        this.profilePhotoURL = user.profilePhotoURL;
    }

}

File: isegoria/src/main/java/com/eulersbridge/isegoria/network/api/models/GenericUser.kt
package com.eulersbridge.isegoria.network.api.models


interface GenericUser {
    var gender: String
    var nationality: String?

    var email: String
    var givenName: String
    var familyName: String

    var institutionId: Long?

    var level: Long
    var experience: Long

    var completedTasksCount: Long
    var completedBadgesCount: Long

    var profilePhotoURL: String?

    val fullName: String
        get() = if (givenName.isBlank() && familyName.isBlank()) {
            ""

        } else if (familyName.isBlank()) {
            givenName

        } else if (givenName.isBlank()) {
            familyName

        } else {
            "$givenName $familyName"
        }
}